### PR TITLE
removed kv_tables_as_object

### PR DESCRIPTION
**Pull request by kimjh2005**
## Change Description

Remove kv_tables_as_object and template specialiszed may_not_exist

## Change Type
**Select ONE**
- [ ] Documentation

- [ ] Stability bug fix

- [ ] Other

- [ ] Other - special case

## Consensus Changes
- [ ] Consensus Changes

## API Changes
- [ ] API Changes

## Documentation Additions
- [ ] Documentation Additions

### CODE REVIEW

File: libraries/chain/include/eosio/chain/abi_def.hpp [hunk 1]
```diff
@@ -222,22 +216,10 @@ void from_variant(const fc::variant& v, eosio::chain::may_not_exist<T>& e) {
...
-template<typename ST, typename T>
-ST& operator >> (ST& s, eosio::chain::kv_tables_as_object<T>& v) {
-   if( s.remaining() )
-      raw::unpack(s, v.value);
-   return s;
-}
-
-template<typename T>
-void to_variant(const eosio::chain::kv_tables_as_object<T>& o, fc::variant& v) {
+//template <typename T, typename = std::enable_if_t<std::is_same_v<T, std::map<eosio::chain::name, eosio::chain::kv_table_def>>, void>>
```
Comment (jeffreyssmith2nd) [inline libraries/chain/include/eosio/chain/abi_def.hpp:219]: Should this be deleted?

File: libraries/chain/include/eosio/chain/abi_def.hpp [hunk 2]
```diff
@@ -260,9 +242,11 @@ void to_variant(const eosio::chain::kv_tables_as_object<T>& o, fc::variant& v) {
    v = variant(mvo);
 }
 
-template<typename T>
-void from_variant(const fc::variant& v, eosio::chain::kv_tables_as_object<T>& o) {
-    EOS_ASSERT( v.is_object(), eosio::chain::invalid_type_inside_abi, "variant is not an variant_object type");
+//template <typename T, typename = std::enable_if_t<std::is_same_v<T, std::map<eosio::chain::name, eosio::chain::kv_table_def>>, void>>
```
Comment (jeffreyssmith2nd) [inline libraries/chain/include/eosio/chain/abi_def.hpp:245]: Should this be deleted?

File: libraries/chain/include/eosio/chain/abi_def.hpp (hunk 1 above)
Comment (linhuang-blockone) [inline libraries/chain/include/eosio/chain/abi_def.hpp:219]: Can you remove this comment?

**Review by linhuang-blockone — APPROVED** [review]

**Comment by kimjh2005** [comment]
Need to re-create PR request.

**Pull request closed without merging**